src/thunderbots/software/ai/hl/stp/play/kickoff_enemy_play.cpp

```diff
@@ -0,0 +1,141 @@
+#include "ai/hl/stp/play/kickoff_enemy_play.h"
+
+#include "ai/hl/stp/evaluation/enemy_threat.h"
+#include "ai/hl/stp/evaluation/possession.h"
+#include "ai/hl/stp/play/play_factory.h"
+#include "ai/hl/stp/tactic/move_tactic.h"
+#include "ai/hl/stp/tactic/shadow_kickoff_tactic.h"
+#include "shared/constants.h"
+
+const std::string KickoffEnemyPlay::name = "KickoffEnemy Play";
+
+std::string KickoffEnemyPlay::getName() const
+{
+    return KickoffEnemyPlay::name;
+}
+
+bool KickoffEnemyPlay::isApplicable(const World &world) const
+{
+    return (world.gameState().isReadyState() | world.gameState().isSetupState()) &&
+           world.gameState().isTheirKickoff();
+}
+
+bool KickoffEnemyPlay::invariantHolds(const World &world) const
+{
+    return !world.gameState().isPlaying();
+}
+
+void KickoffEnemyPlay::getNextTactics(TacticCoroutine::push_type &yield)
+{
+    // TODO: This needs to be a goalie tactic
+    auto lone_goalie_tactic_0 = std::make_shared<MoveTactic>(true);
+
+    // 3 robots assigned to shadow enemies. Other robots will be assigned positions
+    // on the field to be evenly spread out
+    std::vector<std::shared_ptr<ShadowKickoffTactic>> shadow_kickoff_tactics = {
+        std::make_shared<ShadowKickoffTactic>(world.field(), true),
+        std::make_shared<ShadowKickoffTactic>(world.field(), true),
+        std::make_shared<ShadowKickoffTactic>(world.field(), true),
+    };
+
+    // these positions are picked according to the following slide
+    // https://images.slideplayer.com/32/9922349/slides/slide_2.jpg
+    // since we only have 6 robots at the maximum, 3 robots will shadow threats
+    // up front, 1 robot is dedicated as the goalie, and the other 2 robots will defend
+    // either post (as show in the image)
+    //
+    // Positions 1,2 are the most important, 3,4,5 are a fallback
+    // if there aren't as many threats to shadow. Robots will be assigned
+    // to those positions in order of priority
+    //
+    // 		+--------------------+--------------------+
+    // 		|                    |                    |
+    // 		|                    |                    |
+    // 		|                    |                    |
+    // 		+--+                 |  4            2 +--+
+    // 		|  |                 |                 |  |
+    // 		|  |               +-+-+               |  |
+    // 		|  |               |   |             3 |  |
+    // 		|  |               +-+-+               |  |
+    // 		|  |                 |                 |  |
+    // 		+--+                 |  5            1 +--+
+    // 		|                    |                    |
+    // 		|                    |                    |
+    // 		|                    |                    |
+    // 		+--------------------+--------------------+
+
+    std::vector<Point> defense_positions = {
+        Point(world.field().friendlyGoalpostNeg().x() + world.field().goalWidth() +
```

Comment:
DId you mean goal width or defense width here? Your diagram implied defense width, aka the width of the defense area box, not the net itself?

Reply:
ah yes, i wasn't aware of the naming conventions, fixed